### ISSUE TITLE
LSQR on gpu error

### ISSUE THREAD

**Issue by ciba12** [JuliaSmoothOptimizers/Krylov.jl#368]
I am getting the following error when using lsqr() function with CUDA:

```
ERROR: GPU compilation of kernel broadcast_kernel(CUDA.CuKernelContext, CuDeviceVector{Float64, 1}, Base.Broadcast.Broadcasted{Nothing, Tuple{Base.OneTo{Int64}}, typeof(identity), Tuple{Base.Broadcast.Extruded{Array{Float64, 0}, Tuple{}, Tuple{}}}}, Int64) failed
KernelError: passing and using non-bitstype argument

Argument 4 to your kernel function is of type Base.Broadcast.Broadcasted{Nothing, Tuple{Base.OneTo{Int64}}, typeof(identity), Tuple{Base.Broadcast.Extruded{Array{Float64, 0}, Tuple{}, Tuple{}}}}, which is not isbits:
  .args is of type Tuple{Base.Broadcast.Extruded{Array{Float64, 0}, Tuple{}, Tuple{}}} which is not isbits.
    .1 is of type Base.Broadcast.Extruded{Array{Float64, 0}, Tuple{}, Tuple{}} which is not isbits.
      .x is of type Array{Float64, 0} which is not isbits.
```

Code to replicate:
```
using Krylov
using CUDA
Krylov.lsqr(CuSparseMatrixCSC(sparse([1.0 0;0 0])),CuVector([1.0;2]))
```

Julia says the error occurs on the line 116 in `lsqr.jl`:
`  x .= zeros(T)`, shouldn't it be  `  x .= zero(T)` ?

**Comment by ciba12** [comment]
Thanks!

**Comment by amontoison** [comment]
The issue is fixed with the release 0.7.5 ! :smiley:
